setuptools_scm/__init__.py

```diff
@@ -116,8 +117,17 @@ def get_version(root='.',
     root = os.path.abspath(root)
     trace('root', repr(root))
 
+    if clean_tag:
```

Comment:
please put the hack into a monkeypatching context-manager that is ensured to run and can be tested separately